add readme for popup toolkit component (#192)
Esri/arcgis-runtime-toolkit-android: Documentation/Popup/README.md

# PopupView

> The PopupView shows the Popup's attribute list in a Recyclerview, allows user to edit Popup's attribute list and delete the Popup. 

Reviewer: ```suggestion
The PopupView shows a Popup's attribute list in a Recyclerview, allows users to edit a Popup's attribute list and delete the Popup. 
```

Author: updated

## Workflow 

The app can define the `PopupView` in its view hierarchy, as `PopupView` extends FrameLayout. The app 
has to set the `popup` and `popupManager` property on the `PopupView` with the Popup it wasnts to view/edit. The `PopupView` toolkit component
adhers to the `MVVM` pattern and comes with the `PopupViewModel` that exposes the functionality to delete the `GeoElement`, set the edit mode on
the `PopupView` and other LiveData events that the app can listen to, to provide user with UI notifications.

> Here is an example XML code that shows the PopupView and how the PopupView Properties are tied to the PopupViewModel:

Reviewer: ```suggestion
Here is an example XML layout that shows the PopupView and how its properties are bound to the PopupViewModel:
```

Author: updated

                    android:layout_height="0dp"
                    app:layout_constraintBottom_toBottomOf="parent"
                    app:layout_constraintEnd_toEndOf="parent"
                    app:layout_constraintStart_toStartOf="parent"
                    app:layout_constraintTop_toBottomOf="@id/editLayout"
                    app:popup="@{popupViewModel.popup}"
                    app:popupManager="@{popupViewModel.popupManager}"
                    app:editMode="@{popupViewModel.isPopupInEditMode()}">
            </com.esri.arcgisruntime.toolkit.popup.PopupView>
       .....
```

Here is example Kotlin code to set the `Popup` on the `PopupViewModel` and handle the `Events` raised by the `PopupViewModel`

```kotlin

    .....
     identifyLayerResultsFuture.addDoneListener {
                try {
                    val identifyLayerResult = identifyLayerResultsFuture.get()

                    if (identifyLayerResult.popups.size > 0) {
                        popupViewModel.setPopup(identifyLayerResult.popups[0])

     .....
     
     popupViewModel.showSavingProgressEvent.observeEvent(viewLifecycleOwner) { isShowProgressBar ->

>             if (isShowProgressBar) {
>                 requireActivity().window.setFlags(
>                     WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
>                     WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE)
>                 progressBarLayout.visibility = View.VISIBLE
>             } else {
>                 progressBarLayout.visibility = View.GONE
>                 requireActivity().window.clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE)
>             }

Reviewer: For simplicity I suggest...
```suggestion
            if (isShowProgressBar) {
                showProgressBar()
            } else {
                hideProgressBar()
            }
```

Author: updated
